src/java.base/share/classes/java/lang/runtime/EnumInitializer.java

```diff
@@ -0,0 +1,47 @@
+package java.lang.runtime;
+
+import jdk.internal.misc.Unsafe;
+
+import java.lang.invoke.MethodHandle;
+import java.lang.invoke.MethodHandles;
+import java.lang.reflect.Field;
+import java.lang.reflect.Method;
+
+/**
+ * A class for initializing enums.
+ */
+public final class EnumInitializer {
+
+    /**
+     * A method for initializing enums.
+     *
+     * @param enumClassLU the lookup
+     */
+    public static void initializeEnumClass(MethodHandles.Lookup enumClassLU) {
+        if (!enumClassLU.hasFullPrivilegeAccess()) {
```

Comment:
This should also check that `enumClassLU` has the [`ORIGINAL`] lookup mode.

[`ORIGINAL`]: https://docs.oracle.com/en/java/javase/21/docs/api/java.base/java/lang/invoke/MethodHandles.Lookup.html#ORIGINAL